collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java

```diff
@@ -206,22 +216,28 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
     private void performSelectedSearch(int position) {
         hideSearchBox();
         switch(position) {
-            case 0:
+            case BY_NAME_ASC:
```

Comment:
I don't think any of this is needed anymore because all the sort methods do the same thing. I think instead of the switch statement you can just rename `setupAdapter` to `sortWithCurrentlySelectedOrder` and do 
```
saveSelectedSortingOrder(position);
sort();
```